http/codegen/client_cli.go

```diff
@@ -172,61 +88,42 @@ func endpointParser(genpkg string, root *expr.RootExpr, svr *expr.ServerExpr, da
 			Name: sd.Service.PkgName + "c",
 		})
 	}
-	usages := make([]string, len(data))
-	var examples []string
-	for i, cmd := range data {
-		subs := make([]string, len(cmd.Subcommands))
-		for i, s := range cmd.Subcommands {
-			subs[i] = s.Name
-		}
-		var lp, rp string
-		if len(subs) > 1 {
-			lp = "("
-			rp = ")"
-		}
-		usages[i] = fmt.Sprintf("%s %s%s%s", cmd.Name, lp, strings.Join(subs, "|"), rp)
-		if i < 5 {
-			examples = append(examples, cmd.Example)
-		}
+
+	var flagsCode bytes.Buffer
+	err := server.EndpointParserFlagsSection(data).Write(&flagsCode)
```

Comment:
Since this function requires the common `SubcommandData` as argument, we can run a for loop and collect the common `SubcommandData` from the HTTP specific `subcommandData` struct. We can also use the collected common SubcommandData type in other places.